lib/pbench/common/logger.py

```diff
@@ -8,6 +9,35 @@
 from pbench.common.exceptions import BadConfig
 
 
+class HostnameFilter(logging.Filter):
+    """
+    This logging filter allows adding hostname to standard Python log format
+    strings, in the form `%(hostname)s`
+    """
+
+    def __init__(self, hostname: str = None):
+        """
+        Initialize the filter. By default, `%(hostname)s` will be expanded to
+        the platform hostname. This can be overridden by specifying an explicit
+        hostname value.
+
+        NOTE: the `Filter` class is an "example" filter that filters for a
+        logger "name". Calling the superclass constructor adds two instance
+        properties our `filter` will ignore, but LGTM.com complains if it's
+        omitted. In theory we can suppress the warning with
+        `# lgtm[py/missing-call-to-init]` but that doesn't seem to work.
+
+        Args:
+            hostname: Hostname to report. Defaults to platform.node().
+        """
+        super().__init__()
+        self.hostname = hostname if hostname else platform.node()
```

Comment:
In a production environment, the hostname is already in the logs added by `journald` (at least for our target RHEL environments).

While our test environment doesn't use `syslog`, production does and host name is added.

When remote ToolMeisters send logs to the Tool Data Sink, the host name is added there, see https://github.com/distributed-system-analysis/pbench/blob/main/lib/pbench/agent/redis.py#L212.

What is the case where we were missing a host name in the log output?

Reply:
I'm not quite sure what you mean here; but I assume you mean when we're collecting the distributed syslogs into a common place. Where messages do currently show up in local `journalctl` streams (which, actually, none appear to do right now), they wouldn't be annotated ... but of course in that case the context is implicit so we don't need to add anything.

Ah, well; it was a fun experiment.